src/internal_rules.jl

```diff
@@ -761,47 +775,73 @@ function EnzymeRules.augmented_primal(
             end
         end
     end
-    cache = if isa(A, Const)
-        nothing
-    else
-        dfact
-    end
 
+    cache = isa(A, Const) ? nothing : (fact_returned, dfact)
     return EnzymeRules.AugmentedReturn(fact, dfact, cache)
```

Comment:
this should now be fact_returned (aka only returned if EnzymeRules.needs_primal)